airflow/models/dag.py

```diff
@@ -1463,13 +1463,8 @@ def partial_subset(
         """
         # deep-copying self.task_dict and self._task_group takes a long time, and we don't want all
         # the tasks anyway, so we copy the tasks manually later
-        task_dict = self.task_dict
-        task_group = self._task_group
-        self.task_dict = {}
-        self._task_group = None  # type: ignore
-        dag = copy.deepcopy(self)
-        self.task_dict = task_dict
-        self._task_group = task_group
+        memo = {id(self.task_dict): None, id(self._task_group): None}
+        dag = copy.deepcopy(self, memo)  # type: ignore
```

Comment:
Just for my sake of understanding, the args passed to `memo` are copied as it is instead of creating new copy of that object?

Reply:
Should we remove the comments on L1464 & L1465 now?

Reply:
> Just for my sake of understanding, the args passed to memo are copied as it is instead of creating new copy of that object?

Yes. `memo` is used so deepcopy can keep track of what it's already copied (mostly in the case of reference loops in objects etc)

> Should we remove the comments on L1464 & L1465 now or update it to reflect the change?

No, the comment is still appropriate and the copying is done on L1485 and L1513 (unchanged)